File: buerguer-pig/AMarotoChivite-BurguerPig/AMarotoChivite-Aemet-Java/src/main/java/Ficheros/Aemet/models/AemetDailyConsultToExport.java
package Ficheros.Aemet.models;

import com.google.gson.annotations.Expose;
import org.simpleframework.xml.Transient;

public class AemetDailyConsultToExport {

    private final Boolean isPrecipitacion = false;
    private final String date;
    private Double mediaTemperature = 0.0;
    private Temperature maxTemperature;
    private Temperature minTemperature;
    private double precipitation = 0.0;

    @Transient // Para que no me serialice el campo en XML
    @Expose(serialize = false) // Para que no me serialice el campo en GSON
    private transient Integer temperatureCount = 0; // transient delante del tipo para no SERIALIZARLO en general
    @Transient
    @Expose(serialize = false)
    private transient Double temperatureSum = 0.0;
    private boolean precipitacion;

    public AemetDailyConsultToExport(String date) {
        this.date = date;
    }

    public void addTemperature(
            Double maxTemperatureValue,
            String maxTemperatureTime,
            Double minTemperatureValue,
            String minTemperatureTime,
            String lugar
    ) {
        Temperature maxTemp = new Temperature(maxTemperatureValue, maxTemperatureTime, lugar);
        Temperature minTemp = new Temperature(minTemperatureValue, minTemperatureTime, lugar);

        if (maxTemperature == null || maxTemp.getValue() > maxTemperature.getValue()) {
            maxTemperature = maxTemp;
        }

        if (minTemperature == null || minTemp.getValue() < minTemperature.getValue()) {
            minTemperature = minTemp;
        }

        temperatureCount++;
        temperatureSum += maxTemperatureValue + minTemperatureValue;
    }

    public void addPrecipitation(double value) {
        precipitation += value;
    }

    public Double getAverageTemperature() {
        if (temperatureCount > 0) {
            return temperatureSum / (temperatureCount * 2);
        } else {
            return 0.0;
        }
    }

    public void setMediaTemperature(Double mediaTemperature) {
        this.mediaTemperature = mediaTemperature;
    }

    public Double getPrecipitation() {
        return precipitation;
    }

    public void setPrecipitacion(boolean precipitacion) {
        this.precipitacion = precipitacion;
    }
}

File: aemet/AMarotoChivite-Aemet/AMarotoChivite-Aemet-Kotlin/src/main/kotlin/Ficheros/Aemet/models/AemetDailyConsultToExport.kt
package Ficheros.Aemet.models

data class Temperature(
    val value: Double,
    val time: String,
    val lugar: String
)

class AemetDailyConsultToExport(
    val date: String
) {
    var maxTemperature: Temperature? = null
    var minTemperature: Temperature? = null
    var mediaTemperature: Double = 0.0
    var precipitation: Double = 0.0
    var isPrecipitacion: Boolean = false

    @Transient // Transient propio de la librería de kotlin, para que no me serialice el campo
    private var temperatureCount = 0

    @Transient
    private var temperatureSum = 0.0

    fun addTemperature(
        maxTemperatureValue: Double,
        maxTemperatureTime: String,
        minTemperatureValue: Double,
        minTemperatureTime: String,
        lugar: String
    ) {
        val maxTemp = Temperature(maxTemperatureValue, maxTemperatureTime, lugar)
        val minTemp = Temperature(minTemperatureValue, minTemperatureTime, lugar)

        if (maxTemperature == null || maxTemp.value > maxTemperature!!.value) {
            maxTemperature = maxTemp
        }

        if (minTemperature == null || minTemp.value < minTemperature!!.value) {
            minTemperature = minTemp
        }

        temperatureCount++
        temperatureSum += maxTemperatureValue + minTemperatureValue
    }

    fun addPrecipitation(value: Double) {
        precipitation += value
    }

    fun getAverageTemperature(): Double {
        return if (temperatureCount > 0) {
            temperatureSum / (temperatureCount * 2)
        } else {
            0.0
        }
    }
}